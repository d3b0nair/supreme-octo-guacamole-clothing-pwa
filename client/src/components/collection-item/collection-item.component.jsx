import React from "react";
import { connect } from "react-redux";
import { addItem } from "../../redux/cart/cart.actions";

import {
  CollectionItemContainer,
  ImageContainer,
  CollectionFooterContainer,
  ItemName,
  ItemPrice,
  AddToCartButton,
} from "./collection-item.styles";

function CollectionItem({ item, addItem }) {
  const { id, name, price, imageUrl } = item;
  return (
    <CollectionItemContainer key={id}>
      <ImageContainer className="image" url={imageUrl} />
      <CollectionFooterContainer>
        <ItemName>{name}</ItemName>
        <ItemPrice>{price}$</ItemPrice>
      </CollectionFooterContainer>
      <AddToCartButton inverted onClick={() => addItem(item)}>
        Add to cart
      </AddToCartButton>
    </CollectionItemContainer>
  );
}
const mapDispatchToProps = (dispatch) => ({
  addItem: (item) => dispatch(addItem(item)),
});
export default connect(null, mapDispatchToProps)(CollectionItem);
