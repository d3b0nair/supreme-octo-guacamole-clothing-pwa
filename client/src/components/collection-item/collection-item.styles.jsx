import styled from "styled-components";

import CustomButton from "../custom-button/custom-button.component";

import { frostedGlass } from "./../../global.styles";

const getImageUrl = (props) => {
  return props.url;
};

export const CollectionItemContainer = styled.div`
  white-space: nowrap;
  display: flex;
  flex-direction: column;
  height: 350px;
  width: 22vw;
  align-items: center;
  position: relative;
  border-radius: 4px;
  padding: 10px;
  margin: 3px;
  ${frostedGlass}
  &:hover {
    .image {
      opacity: 0.8;
    }
    button {
      opacity: 0.85;
      display: flex;
    }
  }
  @media screen and (max-width: 888px) {
    width: 80vw;
    margin: 0 0 10px 0;
    ${"" /* flex-direction: column-reverse; */}
  }
  @media screen and (max-width: 300px) {
    &:hover {
      button {
        display: block;
      }
    }
  }
`;

export const ImageContainer = styled.div`
  width: 100%;
  height: 95%;
  background-size: cover;
  background-position: center;
  margin-bottom: 5px;
  opacity: 1;
  background-image: url(${getImageUrl});
  border-width: 1px;
  border-style: solid;
  border-color: rgb(255, 255, 255);
`;
export const CollectionFooterContainer = styled.div`
  width: 100%;
  height: 10%;
  display: flex;
  justify-content: space-between;
  font-size: 150%;
`;

export const ItemName = styled.span`
  width: 90%;
  margin-bottom: 15px;
`;
export const ItemPrice = styled.span`
  width: auto;
`;

export const AddToCartButton = styled(CustomButton)`
  width: 80%;
  opacity: 0.7;
  position: absolute;
  top: 235px;
  display: none;
  @media screen and (max-width: 888px) {
    display: block;
    opacity: 1;
  }
`;
AddToCartButton.displayName = "AddToCartButton";
