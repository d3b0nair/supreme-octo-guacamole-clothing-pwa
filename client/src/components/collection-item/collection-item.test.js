import React from "react";
import {
  render,
  fireEvent,
  waitFor,
  screen,
  cleanup,
} from "@testing-library/react";
import { CollectionItem } from "./collection-item.component";

const mockAddItem = jest.fn();
const mockProps = (otherProps) => ({
  ...otherProps,
  item: [],
  addItem: mockAddItem,
});
const component = (mockProps) => <CollectionItem {...mockProps} />;

afterEach(cleanup);

describe("CollectionItem component", () => {
  it("should mount", () => {
    const { container } = render(component(mockProps()));
    expect(container).toMatchSnapshot();
  });
  it("should call addItem on button click", async () => {
    render(component(mockProps()));
    await waitFor(() => fireEvent.click(screen.getByTestId("AddToCartButton")));
    expect(mockAddItem).toHaveBeenCalled();
  });
});
