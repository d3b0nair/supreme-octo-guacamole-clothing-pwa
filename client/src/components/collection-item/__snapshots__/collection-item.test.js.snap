// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem component should mount 1`] = `
<div>
  <div
    class="sc-gsDKAQ cxkvTq"
  >
    <div
      class="sc-dkPtRN fRRZdH image"
    />
    <div
      class="sc-hKwDye mbOMF"
    >
      <span
        class="sc-eCImPb jgGulH"
      />
      <span
        class="sc-jRQBWg bIGsJs"
      >
        $
      </span>
    </div>
    <button
      class="sc-bdvvtL gRKetO sc-gKclnd ePLOnP"
      data-testid="AddToCartButton"
    >
      Add to cart
    </button>
  </div>
</div>
`;
